.circleci/config.yml

```diff
@@ -9,6 +9,7 @@ jobs:
     docker:
       - image: circleci/python:3.6
     working_directory: ~/repo
+    parallelism: 4
```

Comment:
why 4?

Reply:
No particular reason, let me see if I can scale up more

Reply:
There doesnt seem to be a maximum option... Why not just keep it 4 with scope for improving?

Reply:
no but typically we could have it at 7, the number of examples to be run.

Reply:
oh that may not be the best thing, cause every container builds all the environments to just run one test. Thats too much overhead and maybe too much unwanted resource usuage. Right now duration is nearly 23min, maybe I will see how much gain we can expect and then decide..

Reply:
do we have a hard-limit on resources, or is this a purely reasonable concern?

Reply:
Well i think we have a limit of 2,500 free credits/week on circle ci:  https://circleci.com/pricing/#
The way how many credits is used by us, I have no idea really..

Reply:
ok let's not spend too much time on that we can stick with 4, I am fine with it. We should consider switching to GH actions which is clearer and more integrated.

Reply:
I agree